.gitignore

```diff
@@ -48,7 +48,13 @@
 CMakeFiles
 CMakeCache.txt
 cmake_install.cmake
+CMakeFiles/
 /compile_commands.json
+arbor-config-version.cmake
+arbor-config.cmake
+bin/
+arbor/include/arbor/assert_macro.hpp
+ext/gbench-build-prefix/
```

Comment:
This is possibly a very personal preference, but I'd rather we don't fill up the .gitignore file with lots of special cases; ultimately, we can't predict how we're going to build things, or locally install things, etc., and can at best catch only some of the common ones.

So we have to be careful when using git add and git commit anyway. .gitignore can't save us, unless we're typically making non-source files in source directories, which we really shouldn't be doing anyway.

Reply:
Yeah, I'm not sure about it either. Since I saw some lines blocking Cmake stuff I decided to add mine. Should that be removed from `.gitignore` as well?

Reply:
That's my preference, but I think it's a preference that may be in opposition to @bcumming's.

Reply:
Right. I think that I am somewhere in between.

The artifacts generated by cmake during configure and build phases shouldn't be in the `.gitignore`. However, that's easy for me to say, because I think I added `build` (the path in which I do my out of tree CMake) to `.gitignore` a long time ago.

The CMake artifacts currently in .gitignore were probably added a long long time ago without any thought given to long term maintenance of `.gitignore`.

@halfflat, you would probably be in favor of removing `build` from the `.gitignore`? I am not so keen, as it would force me to do completely out of tree builds (because unlike you, I don't like having to be so careful when `git add`ing).

Reply:
To make it clear: yes, let's remove all of those CMake and build artifacts from the `.gitignore` instead of expanding them.

Reply:
Yeah, for example, I personally don't use `build/` as a build directory (I typically use `b/`) so even that entry doesn't help me :)

Reply:
About the `build` thing ... I tried adding a file named `build-catalogue.py` which was stopped outright by `.gitignore` rules